File: chat-backend/src/main/java/model/Message.java
package model;


import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.Generated;
import lombok.NoArgsConstructor;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.Instant;
import java.util.UUID;


@AllArgsConstructor
@Data
@NoArgsConstructor
@Document(collection = "message")
public class Message {

    private String id;

    private String content;

    private Long timestamp;

    private String groupId;

    public Message(String content){
        this.id = UUID.randomUUID().toString();
        this.content = content;
        this.groupId = groupId;
        this.timestamp = Instant.now().getEpochSecond() * 1000;
    }

}

File: chat-backend/src/main/kotlin/com/thm/greenhat/greenhatchat/model/Message.kt
package com.thm.greenhat.greenhatchat.model

//import kotlinx.serialization.*;
import org.bson.types.ObjectId
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import java.time.Instant
import java.util.*

@Document("messages")
data class Message(
        @Id var _id: String = ObjectId.get().toString(),
        var content: String,
        var created:Long = Instant.now().epochSecond * 1000,
        var userId:String,
        var groupId:String
)